### PR TITLE
Update development with changes from master

### PR DESCRIPTION
**Pull request by FWuellhorst**
The master and development are out of sync, which leads to issues in #757 . We should update them separately.

### CODE REVIEW

**Comment by FWuellhorst** [comment]
@DaJansenGit : Most changes are for the new CI structure. However, some files are changed using some new features you implement at some point. Can you check if everything is compatible?

File: tests/test_data.py
Comment (DaJansenGit) [inline tests/test_data.py:3178]: why did you delete this test?

**Review by DaJansenGit — COMMENTED** [review]
Thanks for the good work! There is just one removed test I'm not sure about (see comment)

File: tests/test_data.py
Comment (FWuellhorst) [reply tests/test_data.py:3178]: Not intendet, this test does not exist in the master, maybe a merge conflict. It should be in there.

Comment (DaJansenGit) [reply tests/test_data.py]: Ok, I commented the test now as the code that is needed for test is only in dev and not in this branch. Don't complety understand what went wrong here. But lets merge this into dev after the tests completed and then merge dev -> master. We need to uncomment this test then again by hand before the merge into master.

**Review by DaJansenGit — APPROVED** [review]

**Pull request merged**
